drivers/ieee802154/ieee802154_efr32.c

```diff
@@ -0,0 +1,558 @@
+	 * sent or CCA failed.
+	 */
+    struct k_sem tx_wait;
+
+    /* TX result. Set to 1 on success, 0 otherwise. */
+    bool tx_success;
+
+    u8_t channel;
+
+    RAIL_Handle_t rail_handle;
+};
+
+static struct efr32_context efr32_data;
+
+const RAIL_IEEE802154_Config_t rail_ieee802154_config = {
+    .addresses = NULL,
+    .ackConfig = {
+        .enable = true,
+        .ackTimeout = 894,
+        .rxTransitions = {
+            RAIL_RF_STATE_RX,
+            RAIL_RF_STATE_RX,
+        },
+        .txTransitions = {
+            RAIL_RF_STATE_RX,
+            RAIL_RF_STATE_RX,
+        }},
+    .timings = {
+        .idleToRx = 100,         
+        .idleToRx = 192 - 10,    
+        .idleToTx = 100,         
+        .rxToTx = 192,           
+        .rxSearchTimeout = 0,    
+        .txToRxSearchTimeout = 0,
+    },
+    .framesMask = RAIL_IEEE802154_ACCEPT_STANDARD_FRAMES, 
+    .promiscuousMode = false,                             
+    .isPanCoordinator = false,                            
+};
+
+CORE_irqState_t CORE_EnterCritical(void)
+{
+    LOG_DBG("Enter crit");
+    return irq_lock();
+}
+
+void CORE_ExitCritical(CORE_irqState_t irqState)
+{
+    (void)irqState;
+    irq_unlock(irqState);
+}
+
+CORE_irqState_t CORE_EnterAtomic(void)
+{
+    LOG_DBG("Enter atomic");
+    return irq_lock();
+}
+
+void CORE_ExitAtomic(CORE_irqState_t irqState)
+{
+    (void)irqState;
+    irq_unlock(irqState);
+}
+
+static void efr32_rail_cb(RAIL_Handle_t rail_handle, RAIL_Events_t a_events);
+
+static RAIL_Config_t s_rail_config = {
+    .eventsCallback = &efr32_rail_cb,
+    .protocol = NULL,
+    .scheduler = NULL,
+};
+
+static inline u8_t *get_mac(struct device *dev)
+{
+    uint64_t uniqueID = SYSTEM_GetUnique();
+    uint8_t *mac = (uint8_t *)&uniqueID;
+    return mac;
+}
+
+static void efr32_iface_init(struct net_if *iface)
+{
+    struct device *dev = net_if_get_device(iface);
+    struct efr32_context *efr32 = dev->driver_data;
+    u8_t *mac = get_mac(dev);
+
+    net_if_set_link_addr(iface, mac, 8, NET_LINK_IEEE802154);
+    efr32->iface = iface;
+    ieee802154_init(iface);
+}
+
+static enum ieee802154_hw_caps efr32_get_capabilities(struct device *dev)
+{
+    return IEEE802154_HW_FCS |
+           IEEE802154_HW_2_4_GHZ |
+           IEEE802154_HW_FILTER |
+           IEEE802154_HW_TX_RX_ACK;
+}
+
+static int efr32_cca(struct device *dev)
+{
+    LOG_DBG("Doing CCA");
+}
+
+static int efr32_set_channel(struct device *dev, u16_t channel)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_PrepareChannel(efr32->rail_handle, channel);
+
+    if (status != RAIL_STATUS_NO_ERROR)
+    {
+        return -EIO;
+    }
+
+    efr32->channel = channel;
+
+    return 0;
+}
+
+static int efr32_set_pan_id(struct device *dev, u16_t pan_id)
+{
+    struct efr32_context *efr32 = dev->driver_data;
+
+    RAIL_Status_t status;
+    status = RAIL_IEEE802154_SetPanId(efr32->rail_handle, pan_id, 0);
```

Comment:
directly test the function, no need of the status variable.  Apply that everywhere where relevant